crates/types/src/rpc/transactions.rs

```diff
@@ -0,0 +1,342 @@
+            Transaction::L1Handler(tx) => tx.get_transaction_hash(),
+            Transaction::DeployAccount(tx) => tx.get_transaction_hash(),
+            Transaction::Invoke(tx) => tx.get_transaction_hash(),
+            Transaction::Deploy(tx) => tx.get_transaction_hash(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum DeclareTransaction {
+    Version0(DeclareTransactionV0V1),
+    Version1(DeclareTransactionV0V1),
+    Version2(DeclareTransactionV2),
+}
+
+impl DeclareTransaction {
+    pub fn get_max_fee(&self) -> Fee {
+        match self {
+            DeclareTransaction::Version0(tx) => tx.get_max_fee(),
+            DeclareTransaction::Version1(tx) => tx.get_max_fee(),
+            DeclareTransaction::Version2(tx) => tx.get_max_fee(),
+        }
+    }
+
+    pub fn get_transaction_hash(&self) -> &TransactionHash {
+        match self {
+            DeclareTransaction::Version0(tx) => tx.get_transaction_hash(),
+            DeclareTransaction::Version1(tx) => tx.get_transaction_hash(),
+            DeclareTransaction::Version2(tx) => tx.get_transaction_hash(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, Eq, PartialEq, Deserialize, Serialize)]
+pub struct InvokeTransactionV0 {
+    pub transaction_hash: TransactionHash,
+    pub max_fee: Fee,
+    pub version: TransactionVersion,
+    pub signature: TransactionSignature,
+    pub nonce: Nonce,
+    pub contract_address: ContractAddress,
+    pub entry_point_selector: EntryPointSelector,
+    pub calldata: Calldata,
+}
+
+impl InvokeTransactionV0 {
+    pub fn get_max_fee(&self) -> Fee {
+        self.max_fee
+    }
+
+    pub fn get_transaction_hash(&self) -> &TransactionHash {
+        &self.transaction_hash
+    }
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum InvokeTransaction {
+    Version0(InvokeTransactionV0),
+    Version1(InvokeTransactionV1),
+}
+
+impl InvokeTransaction {
+    pub fn get_max_fee(&self) -> Fee {
+        match self {
+            InvokeTransaction::Version0(tx) => tx.get_max_fee(),
+            InvokeTransaction::Version1(tx) => tx.get_max_fee(),
+        }
+    }
+
+    pub fn get_transaction_hash(&self) -> &TransactionHash {
+        match self {
+            InvokeTransaction::Version0(tx) => tx.get_transaction_hash(),
+            InvokeTransaction::Version1(tx) => tx.get_transaction_hash(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Default, Eq, PartialEq, Deserialize, Serialize)]
+pub struct L1HandlerTransaction {
+    pub transaction_hash: TransactionHash,
+    pub version: TransactionVersion,
+    pub nonce: Nonce,
+    pub contract_address: ContractAddress,
+    pub entry_point_selector: EntryPointSelector,
+    pub calldata: Calldata,
+}
+
+impl L1HandlerTransaction {
+    pub fn get_max_fee(&self) -> Fee {
+        Fee(0)
+    }
+
+    // TODO: if tyoe implementing copy shall be returned by reference?
```

Comment:
typo -> "tyoe"

Reply:
I'm not a fan of leaving TODO in code on main branch, we can easily lose track of all those TODOs. I would address that in github issue.

Reply:
Removed this since I've decided to return by reference. All getters except the one's that return primitive shall return `const reference`